File: 1552-build-an-array-with-stack-operations/1552-build-an-array-with-stack-operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> operations; // To store the operations

        int i = 0; // Pointer for the target array
        int num = 1; // The number in the range [1, n]

        while (i < target.size() && num <= n) {
            // If the current target element matches the number, perform a "Push" operation.
            if (target[i] == num) {
                operations.push_back("Push");
                i++;
            } else {
                // If they don't match, perform a "Push" followed by a "Pop" operation.
                operations.push_back("Push");
                operations.push_back("Pop");
            }
            num++; // Move to the next number
        }

        return operations;
    }
};

File: 1552-build-an-array-with-stack-operations/1552-build-an-array-with-stack-operations.py
class Solution:
    def buildArray(self, target, n):
        result = []
        index = 0  # Initialize an index to keep track of the current target element.

        for i in range(1, n + 1):
            if index >= len(target):
                break  # If we have processed all target elements, exit the loop.
            
            result.append("Push")
            
            if target[index] == i:
                index += 1  # Move to the next target element when they match.
            else:
                result.append("Pop")  # If the elements don't match, simulate Pop.

        return result


# # Example usage:
# solution = Solution()
# target = [1, 3]
# n = 3
# stack_operations = solution.buildArray(target, n)
# print(stack_operations)
